pkg/provider/azure_controller_common.go

```diff
@@ -231,6 +237,7 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, async bool, diskName,
 		resourceGroup = c.resourceGroup
 	}
 
+	waitForBatch = true
```

Comment:
if `c.attachDiskProcessor.Do()` returns an error, the `attachDiskBatchToNode` would not be called and the channel never closed. You should only set `waitForBatch` to `true` if this method return no error.

Reply:
Ah good catch. Thanks!